tmpl/wiki/error-handling.md

```diff
@@ -0,0 +1,384 @@
+[ This DRAFT document is intended to explain how error handling should work in Mirage.
+  Currently, it's just my (Thomas Leonard's) opinions.
+  Please add comments on the bits you disagree with.
+]
+
+#### Goals
+
+POSIX and C generally make for very poor error handling:
+the information reported to the user is just a small number stored in `errno` and
+errors are easily ignored accidentally.
+
+With Mirage, we have the opportunity to provide something much better.
+OCaml's structured types and exceptions can provide rich diagnostic information that is readable to humans and to machines,
+and its exhaustiveness checks mean we can force callers to consider errors where necessary.
+
+There are three main aspects to consider:
+
+- Providing diagnostic information to humans.
```

Comment:
worth distinguishing whether this is to be consumed at the time (standard kind of debugging backtrace) vs post hoc analysis from logs?